cpp/src/lists/copying/slice.cu

```diff
@@ -0,0 +1,68 @@
+#include <thrust/iterator/counting_iterator.h>
+#include <cudf/detail/copy_range.cuh>
+#include <cudf/detail/gather.cuh>
+#include <cudf/lists/lists_column_view.hpp>
+#include <iostream>
+
+namespace cudf {
+namespace lists {
+namespace detail {
+
+// new lists column from a subset of a lists_column_view
+std::unique_ptr<cudf::column> slice(lists_column_view const& lists,
+                                    size_type start,
+                                    size_type end,
+                                    cudaStream_t stream,
+                                    rmm::mr::device_memory_resource* mr)
+{
+  size_type lists_count = lists.size();
+  if (lists_count == 0) { return cudf::empty_like(lists.parent()); }
+  if (end < 0 || end > lists_count) end = lists_count;
```

Comment:
Do we need to be this flexible? Treat all negative values of `end` the same?

Reply:
Borrowed pattern from strings [`slice`](https://github.com/rapidsai/cudf/blob/branch-0.15/cpp/src/strings/copying/copying.cu#L31). Happy to throw here instead.